core/state_processor_rip7560.go

```diff
@@ -118,10 +151,28 @@ func handleRip7560Transactions(transactions []*types.Transaction, index int, sta
 		}
 
 		statedb.SetTxContext(tx.Hash(), index+i)
-
+		beforeValidationSnapshotId := statedb.Snapshot()
 		vpr, vpe := ApplyRip7560ValidationPhases(chainConfig, bc, coinbase, gp, statedb, header, tx, cfg)
 		if vpe != nil {
-			return nil, nil, nil, vpe
+			if skipInvalid {
+				log.Error("Validation failed during block building, should not happen, skipping transaction", "error", vpe)
+				debugInfo := &types.Rip7560TransactionDebugInfo{
+					TxHash:           tx.Hash(),
+					RevertData:       vpe.Error(),
+					FrameReverted:    false,
+					RevertEntityName: "n/a",
+				}
+				validationFailureInfos = append(validationFailureInfos, debugInfo)
+				var vpeCast *ValidationPhaseError
+				if errors.As(vpe, &vpeCast) {
```

Comment:
can it ever be anything else?
better fully initialize the struct in here, and in "else" initialize to to some "defaults" - not "overrride" and fill some fields.

Reply:
I also think "TransactionErrorInfo" struct should hold the error and hash.

Reply:
> can it ever be anything else?

Unfortunately, yes, there are many code paths that end up with a simple `return err`.

I tried wrapping most of then in a `ValidationPhaseError` but this is a tedious task and for now I don't know if it is worth it, the only place where `ValidationPhaseError` is actively used is decoding the revert reason.